source/RoslynAnalyzers/MessageContractAnalyzers.cs

```diff
@@ -94,24 +101,37 @@ void CheckNode(SyntaxNodeAnalysisContext context)
                 EventTypes_MustBeNamedCorrectly(context, typeDec);
             }
             // no specific checks for resources at this point
+            
+            // now check properties
+            CheckProperties(context, typeDec, isExperimental, isMessageType);
         }
 
-        static bool CheckMessageTypeProperties(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDec)
+        static bool CheckProperties(SyntaxNodeAnalysisContext context, TypeDeclarationSyntax typeDec, bool isExperimental, bool isMessageType)
         {
             var result = true;
             foreach (var propDec in typeDec.DescendantNodes().OfType<PropertyDeclarationSyntax>())
             {
-                // only validate public properties
-                if (!propDec.Modifiers.Any(m => m.IsKind(SyntaxKind.PublicKeyword))) continue;
-
-                var required = GetRequiredState(propDec);
-
-                // if anything returns false, propagate false outwards
-                result &= PropertiesOnMessageTypes_MustBeMutable(context, propDec);
-                result &= RequiredPropertiesOnMessageTypes_MustNotBeNullable(context, propDec, required);
-                result &= OptionalPropertiesOnMessageTypes_MustBeInitializedOrNullable(context, propDec, required);
-                result &= PropertiesOnMessageTypes_MustHaveAtLeastOneValidationAttribute(context, propDec, required);
-                result &= SpaceIdPropertiesOnMessageTypes_MustBeOfTypeSpaceId(context, propDec);
+                // generic checks across all api surface types
+                result &= WeDoNotUseEnums_InOurApiSurface(context, propDec, isExperimental);
+                
+                // more detailed checks for request/command/response
+                if (isMessageType)
+                {
+                    var isPublic = PropertiesOnMessageTypes_MustBePublic(context, propDec);
+
+                    if (!isPublic) continue;
+                    // the rest of the checks only apply to public properties.
+                    // if someone were to suppress the public warning because they really needed a protected property, then all the rules don't apply
+
+                    var required = GetRequiredState(propDec);
+                    // if anything returns false, propagate false outwards
+                    result &= isPublic;
```

Comment:
This still confuses me :S... I think I prefer explicitness over terseness here

Reply:
You need to write more C code :smile:

it's the same as `result = result && isPublic` or if you want to be very explicit `if(!isPublic) result = false;`

I'll have a think about how to model it differently. `if(!isPublic) result = false;` is nice enough, but it doesn't go well when applying that pattern to all the lines which follow